backend/users/signals.py

```diff
@@ -0,0 +1,20 @@
+import os
+
+from django.conf import settings
+from django.db.models.signals import post_save, pre_save
+from django.dispatch import receiver
+
+from users.models import get_file_path, Profile, User
+
+
+@receiver(post_save, sender=User)
+def create_profile_signal(sender, instance, created, **kwargs):
+    if created:
+        Profile.objects.create(user=instance)
+
+
+@receiver(pre_save, sender=Profile)
+def remove_old_image(sender, instance, **kwargs):
+    file_path = get_file_path(instance, instance.file)
+    if os.path.exists(os.path.join(settings.MEDIA_ROOT, file_path)):
+        os.remove(os.path.join(settings.MEDIA_ROOT, file_path))
```

Comment:
No need to use `get_file_path` here, because you can access photo path using `instance.file.path`

```suggestion
def remove_old_image(sender, instance, **kwargs):
    file_path = instance.file.path
    if os.path.exists(file_path):
        os.remove(file_path)
```

Reply:
There's one major problem with current implementation. Even when user updates field on profile model unrelated to image, the image will be deleted anyway. One way to avoid this, is to check if `"file"` is in `update_fields` kwarg passed when signal is invoked. Downside of that approach is that you always have to call `profile.save(update_fields=[...])`.
Second way is creating separate model for image and connecting `pre_save` to it.